src/runtime_src/core/common/query_requests.h

```diff
@@ -919,6 +920,23 @@ struct kds_scu_info : request
   get(const device*) const = 0;
 };
 
+
+struct hw_context_info : request
+{
+  struct data {
+    std::string xclbin_uuid;
+    kds_cu_info::result_type pl_compute_units;
+    kds_scu_info::result_type ps_compute_units;
```

Comment:
Hmm, no suggestion here, but this seems a bit constrained.   Not all hw_context have pl cus or ps cus.

Reply:
The idea was if there are no ps or pl cus in a hardware context we just leave those fields empty. When a report iterates through these vectors they is parsed into the json, so if its empty no parsing! If AIE differs from PL compute units we can add a new vector for it or try to combine them all compute units into one. Might get a bit messy though.

Reply:
Ok, so be it.  Please comment this struct and explain what it is for what the usage you just explained.